crates/api/src/arc.rs

```diff
@@ -0,0 +1,46 @@
+//! Kitsune2 arcs represent a range of hash locations on the DHT.
+//!
+//! An arc can exist in its own right and refer to a range of locations on the DHT.
+//! It can also be used in context, such as an agent. Where it represents the range of locations
+//! that agent is responsible for.
+
+/// The definition of an arc for Kitsune.
+///
+/// The lower bound is inclusive, the upper bound is exclusive.
+pub type ArcLiteral = (u32, u32);
+
+/// A full arc is represented by `(0, u32::MAX)`.
+pub const ARC_LITERAL_FULL: ArcLiteral = (0, u32::MAX);
```

Comment:
How can the upper bound be exclusive? The location *at* u32::MAX is not covered by this "full" arc.

Reply:
That's a mistake. I ran into the top of the range not being covered elsewhere and I think I have this mostly sorted but this is an oversight. I'll take a look but probably focus on the musing below first :)

Reply:
Changed and added tests

Reply:
Superseded by arc PR